### DISCUSSION TITLE
Opacity only works for 0 and 1 for Image with SvgImage Source and Svg control

### DISCUSSION THREAD

**Discussion by darinkes** [wieslawsoltes/Svg.Skia#82] [General]
I noticed you can't fade away an Image with SvgImage Source.

Not sure if this is a limitation of Svg.Skia or Svg itself.

```xaml
<Style Selector="ContentPresenter">
  <Setter Property="Transitions">
    <Transitions>
      <DoubleTransition Property="Opacity" Duration="0:0:0.5" Easing="LinearEasing"/>
    </Transitions>
  </Setter>
</Style>
```

ContentPresenter includes a TextBlock and an Image. Only the TextBlock gets slowly invisible.
The Image only disapears as soon as Opacity becomes 0.0

**Comment by wieslawsoltes** [comment]
@darinkes Could you try with new `Svg` control:
https://github.com/wieslawsoltes/Svg.Skia/blob/706d968a28606bf90621df385b77b4608a91f1c5/samples/AvaloniaSvgSkiaSample/MainWindow.axaml#L22

**Comment by adossey** [comment]
I'm seeing the same behavior as @darinkes using both the Image and Svg controls in Avalonia.Svg.Skia version 0.10.6.1.

**Comment by wieslawsoltes** [comment]
This is an issue in AvaloniaUI with how opacity is blended when using custom draw operation

![rider64_bvo7LHrGJm](https://user-images.githubusercontent.com/2297442/121486119-23bf0080-c9d1-11eb-85ca-d6b5564e89c3.png)

**Comment by wieslawsoltes** [comment]
I have hack fix for that, but not really want it to use https://github.com/wieslawsoltes/Svg.Skia/pull/79

**Comment by adossey** [comment]
Thanks for the quick hack! I tried it out, and it works now when the Opacity is applied directly to the Svg control, but it still doesn't work when the Opacity is inherited from the parent control. I can work around this by just duplicating a little code, but inheriting the Opacity from the parent would certainly be ideal.

**Comment by wieslawsoltes** [comment]
> Thanks for the quick hack! I tried it out, and it works now when the Opacity is applied directly to the Svg control, but it still doesn't work when the Opacity is inherited from the parent control. I can work around this by just duplicating a little code, but inheriting the Opacity from the parent would certainly be ideal.

Without Avalonia fixing this in core library this hack is best I can do.

**Comment by darinkes** [comment]
@wieslawsoltes thanks for taking a look at it!

Should we open an issue at Avalonia?
May I ask you to do it, since you can be more specific what needs to be done?

**Comment by wieslawsoltes** [comment]
@darinkes It is known issue, no need to create new one

**Comment by mysteryx93** [comment]
One year later, Avalonia v0.10.15, Avalonia.Svg.Skia v0.10.14 -- I still have that issue.

Opacity is not respected.

    <Svg Path="/Assets/plus.svg" Opacity=".5" />

**Comment by ApaulMyLittleAirport** [comment]
There is still this problem, can it be solved?

**Comment by kevin100702** [comment]
There is still this problem
`<Button Grid.Column="0" Width="40">
				<Image Source="{SvgImage /Assets/Images/monitor.svg}" Opacity="0.6"/>
			</Button>`

**Reply by mysteryx93** [reply]
Personally I worked around the solution by embedding vector icons into font files. Actually very easy to do, and it works smoothly.

**Comment by jamsoft** [comment]
I had a very specific use case ... so I just opened the SVG file and set
```xml
<svg opacity="0.3">
...
</svg>
```

**Comment by XDgl233** [comment]
Another workaround I am using is the DrawingImage method. You can expose the opacity and also the GeometryDrawing brush to let you set the opacity and the fill color for SVG as well. To get the geometry data, you can use [the author's tool](https://github.com/wieslawsoltes/SvgToXaml) to help you get them into xaml code.
```xaml
<Image Opacity="value">
	<DrawingImage>
		<DrawingImage.Drawing>
			<DrawingGroup ClipGeometry="M0,0 V100 H100 V0 H0 Z">
				<GeometryDrawing Brush="#FF000000" Geometry="SVG d" />
				<GeometryDrawing Brush="#FF000000" Geometry="SVG d" />
				...
			</DrawingGroup>
		</DrawingImage.Drawing>
	</DrawingImage>
</Image>
```

**Comment by maxkatz6** [comment]
To fix this issue, add `.With(new SkiaOptions { UseOpacitySaveLayer = true });` in your AppBuilder configuration.

**Reply by leonidcan** [reply]
Is it available in 0.10...?

**Reply by x7ikmet** [reply]
thanks its working now

**Comment by wieslawsoltes** [comment]
add .With(new SkiaOptions { UseOpacitySaveLayer = true }); in your AppBuilder configuration

**Reply by rubiomiguel06** [reply]
I have enabled `UseOpacitySaveLayer`, but seems that Transitions still won't work. For example:
```xaml
<MySkiaControl Opacity="{Binding ViewModelOpacity}">
   <MySkiaControl.Transitions>
      <Transitions>
          <DoubleTransition Property="Opacity"  Duration="0:0:3"/>
       </Transitions>
    </MySkiaControl.Transitions>
</MySkiaControl>
```

Am I missing something?

**Comment by ds1709** [comment]
You can also use `RenderOptions.RequiresFullOpacityHandling="true"` in xaml.
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:i="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:svg="using:Avalonia.Svg"
        xmlns:skiasvg="using:Avalonia.Svg.Skia"
        Width="640"
        Height="480"
        Title="AvaloniaApplication1">
  <Grid RowDefinitions="Auto * *" ColumnDefinitions="* *">
    <Slider Name="PART_OpacitySlider" Grid.Row="0" Grid.Column="0" Grid.ColumnSpan="2" Minimum="0" Maximum="1" Value="1" />
    <Image Grid.Row="1" Grid.Column="0" Opacity="{Binding #PART_OpacitySlider.Value}" RenderOptions.RequiresFullOpacityHandling="true">
      <svg:SvgImage Source="D:\TEMP\logo.svg" />
    </Image>
    <Image Grid.Row="1" Grid.Column="1" Opacity="{Binding #PART_OpacitySlider.Value}" RenderOptions.RequiresFullOpacityHandling="true">
      <skiasvg:SvgImage Source="D:\TEMP\logo.svg" />
    </Image>
    <Image Grid.Row="2" Grid.Column="0" Opacity="{Binding #PART_OpacitySlider.Value}">
      <svg:SvgImage Source="D:\TEMP\logo.svg" />
    </Image>
    <Image Grid.Row="2" Grid.Column="1" Opacity="{Binding #PART_OpacitySlider.Value}">
      <skiasvg:SvgImage Source="D:\TEMP\logo.svg" />
    </Image>
  </Grid>
</Window>
```
<img width="547" height="556" alt="image" src="https://github.com/user-attachments/assets/207f1716-0876-4927-a59f-07e88333e96e" />
